ff/src/biginteger/macros.rs

```diff
@@ -166,11 +166,45 @@ macro_rules! bigint_impl {
                 res
             }
 
+            fn from_bits_le(bits: &[bool]) -> Self {
+                let mut res = Self::default();
+                let mut acc: u64 = 0;
+
+                let bits = bits.to_vec();
+                for (i, bits64) in bits.chunks(64).enumerate() {
+                    for bit in bits64.iter().rev() {
+                        acc <<= 1;
+                        acc += *bit as u64;
+                    }
+                    res.0[i] = acc;
+                    acc = 0;
+                }
+                res
+            }
+
+            #[inline]
+            fn to_bits_be(&self) -> Vec<bool> {
+                BitIteratorBE::new(self.0).collect::<Vec<_>>()
+            }
+
+            #[inline]
+            fn to_bits_le(&self) -> Vec<bool> {
+                BitIteratorLE::new(self.0).collect::<Vec<_>>()
+            }
+
+            #[inline]
+            fn to_bytes_be(&self) -> Vec<u8> {
+                let mut le_bytes = self.to_bytes_le();
+                le_bytes.reverse();
+                le_bytes
+            }
```

Comment:
I think these can all be made default impls on the trait itself? For the bits, you can call `BitIteratorXX::new(self)...`, and `to_bytes_be` is implemented directly in terms of `to_bytes_le`.

Reply:
Sure, but do we want to support BigInteger as a trait for others to be implementing? I assumed BigInt was something internal to the library

Reply:
True, ok I'll do that

Reply:
I didn't realize the trait had `AsRef<[u64]>`, seems kind of word to be in the trait imo. Why would we not allow float back ends?

Reply:
Actually I think its weird to only implement one of `to_bytes_le` and `to_bytes_be` as the default, I'd prefer it in the macro. (The reason to put it in the default impl would be for users implementing these in an alternate fashion, but that then assumes that `to_bytes_le` is the native endianness to do it)

Reply:
Actually I think we can move `to_bytes_le` also to the trait, as it can be implemented from `AsRef<[u64]>`.

(This is an example of why the `AsRef<[u64]>` trait bound is useful)

Reply:
I'll move it for now, but I think its worth revisiting if BigInt needs AsRef<[u64]>, I feel like we shouldn't make PrimeField / elliptic curves depend on that

Reply:
We lose capacity hints with the `to_bytes` being a default implementation, not really sure its worth making it default then?

Reply:
Ah no, I was wrong. The capacity hints are still available